acceptance/input_paths_manager.go

```diff
@@ -0,0 +1,183 @@
+	envPreviousPackPath         = "PREVIOUS_PACK_PATH"
+	envPreviousPackFixturesPath = "PREVIOUS_PACK_FIXTURES_PATH"
+	envLifecyclePath            = "LIFECYCLE_PATH"
+	envPreviousLifecyclePath    = "PREVIOUS_LIFECYCLE_PATH"
+)
+
+type InputPathsManager struct {
+	packPath                 string
+	previousPackPath         string
+	previousPackFixturesPath string
+	lifecyclePath            string
+	previousLifecyclePath    string
+}
+
+func NewInputPathsManager() (*InputPathsManager, error) {
+	packPath := os.Getenv(envPackPath)
+	previousPackPath := os.Getenv(envPreviousPackPath)
+	previousPackFixturesPath := os.Getenv(envPreviousPackFixturesPath)
+	lifecyclePath := os.Getenv(envLifecyclePath)
+	previousLifecyclePath := os.Getenv(envPreviousLifecyclePath)
+
+	inputPaths := []string{
+		packPath,
+		previousPackPath,
+		previousPackFixturesPath,
+		lifecyclePath,
+		previousLifecyclePath,
+	}
+
+	for idx, inputPath := range inputPaths {
+		if inputPath == "" {
+			continue
+		}
+		updatedInputPath, err := filepath.Abs(inputPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "getting absolute path for %s", inputPath)
+		}
+		inputPaths[idx] = updatedInputPath
+	}
+
+	return &InputPathsManager{
+		packPath:                 inputPaths[0],
+		previousPackPath:         inputPaths[1],
+		previousPackFixturesPath: inputPaths[2],
+		lifecyclePath:            inputPaths[3],
+		previousLifecyclePath:    inputPaths[4],
+	}, nil
+}
+
+func (m *InputPathsManager) FillInRequiredPaths(c runCombo) error {
+	githubAssetFetcher, err := NewGithubAssetFetcher()
+	if err != nil {
+		return errors.Wrap(err, "initializing GitHub asset fetcher")
+	}
+
+	if (c.Pack == "previous" || c.PackCreateBuilder == "previous") && m.previousPackPath == "" {
+		fmt.Printf("run combination %+v requires %s to be set\n", c, style.Symbol(envPreviousPackPath))
+
+		version, err := githubAssetFetcher.FetchReleaseVersion("buildpacks", "pack", 0)
+		if err != nil {
+			return err
+		}
+
+		assetDir, err := githubAssetFetcher.FetchReleaseAsset(
+			"buildpacks",
+			"pack",
+			version,
+			packBinaryExp(),
+			true,
+		)
+		if err != nil {
+			return err
+		}
+		assetPath := filepath.Join(assetDir, packBinaryName())
+
+		fmt.Printf("using %s for previous pack path\n", assetPath)
+		m.previousPackPath = assetPath
+	}
+	if (c.Pack == "previous" || c.PackCreateBuilder == "previous") && m.previousPackFixturesPath == "" {
+		fmt.Printf("run combination %+v requires %s to be set\n", c, style.Symbol(envPreviousPackFixturesPath))
+
+		version, err := githubAssetFetcher.FetchReleaseVersion("buildpacks", "pack", 0)
+		if err != nil {
+			return err
+		}
+
+		sourceDir, err := githubAssetFetcher.FetchReleaseSource("buildpacks", "pack", version)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("using %s for previous pack fixtures path\n", sourceDir)
+		m.previousPackFixturesPath = sourceDir // TODO: fix; this is the top level directory - need to traverse directory
```

Comment:
TBD - this is complicated by the fact that the source has a top-level directory that is versioned by commit, e.g. `buildpacks-pack-d42c384`. In theory there would only ever be one directory so it's probably fine.